security/security/src/main/java/io/helidon/security/Security.java

```diff
@@ -1100,10 +1108,47 @@ private boolean notReservedProviderKey(Config config) {
 
         /**
          * Check whether any provider is configured.
+         * @param providerClass type of provider of interest (can be {@link io.helidon.security.spi.AuthenticationProvider} and
+         *                      other interfaces implementing {@link io.helidon.security.spi.SecurityProvider})
          * @return {@code true} if no provider is configured, {@code false} if there is at least one provider configured
          */
-        public boolean noProvider() {
+        public boolean noProvider(Class<? extends SecurityProvider> providerClass) {
+            if (providerClass.equals(AuthenticationProvider.class)) {
+                return atnProviders.isEmpty();
+            }
+            if (providerClass.equals(AuthorizationProvider.class)) {
+                return atzProviders.isEmpty();
+            }
+            if (providerClass.equals(OutboundSecurityProvider.class)) {
+                return outboundProviders.isEmpty();
+            }
+            if (providerClass.equals(AuditProvider.class)) {
+                return auditProviders.isEmpty();
+            }
+            if (providerClass.equals(SubjectMappingProvider.class)) {
+                return subjectMappingProvider == null;
+            }
```

Comment:
`subjectMappingProvider.isEmpty()`

Reply:
It is not `Optional` in `Builder`, only in `Security`. This method is on `Builder`

Reply:
My bad sorry